test/azmi-main-tests/blob/ListBlobs-Tests.cs

```diff
@@ -0,0 +1,193 @@
+            [Theory]
+            [InlineData("container")]
+            [InlineData("prefix")]
+            [InlineData("exclude")]
+            public void TestRealArguments(string argName)
+            {
+                // Arrange
+                var subCommand = new ListBlobs();
+                // Act
+                var arguments = subCommand.Definition().arguments;
+                // Assert
+                Assert.NotEmpty(arguments.Where(a => a.name == argName));
+            }
+
+            [Fact]
+            public void NoFakeArgument()
+            {
+                var subCommand = new ListBlobs();
+                var arguments = subCommand.Definition().arguments;
+                Assert.Empty(arguments.Where(a => a.name == "fakeArgument"));
+            }
+        }
+
+
+        public class GenericExecute_TestsGroup
+        {
+            private readonly Uri _anyValidURL = new Uri("https://www.example.com");
+            private readonly string _failMsg = "Cannot convert input object to proper class";
+            private readonly Exception _testException = new Exception("testing exception");
+
+
+            // test if it will call real execute method
+            [Fact]
+            public void FailsWithNonExistingProperty()
+            {
+                var obj = new { nonExistingProperty = 1 };
+                var subCommand = new ListBlobs();
+
+                // it throws exception
+                var actualExc = Assert.Throws<AzmiException>(() =>
+                   subCommand.Execute(obj)
+                );
+                Assert.Equal(_failMsg, actualExc.Message);
+            }
+
+            [Fact]
+            public void FailsInsideWithGoodProperties()
+            {
+                // Arrange
+                var obj = new ListBlobs.AzmiArgumentsClass
+                {
+                    container = _anyValidURL,
+                    prefix = null,
+                    exclude = null,
+                    identity = null
+                };
+
+                var containerSubstitute = Substitute.For<IContainerClient>();
+                containerSubstitute.GetBlobs(null).Throws(_testException);
+                var subCommand = new ListBlobs(containerSubstitute);
+
+                // Act & Assert
+                var actualExc = Assert.Throws<AzmiException>(
+                    () => subCommand.Execute(obj)
+                );
+                Assert.NotEqual(_failMsg, actualExc.Message);
+                Assert.Equal(_testException.Message, actualExc.InnerException.Message);
+            }
+        }
+
+
+        //
+        // main tests
+        //
+
+        public class ListBlobsExecute_TestsGroup
+        {
+
+            private readonly Uri _anyValidURL = new Uri("https://www.example.com");
+            private static readonly Pageable<BlobItem> _nullBlobItems = null;
+            //private static readonly BlobItem emptyBlob = new BlobItem() { Name = "testBlob" };
```

Comment:
Intentional?

Reply:
Yes, mostly it is dependency on FileSystem which should not be there. But, it will take me at least this much more code in order to fix that and real tests with mocks. Therefore, I opened next ticket to split work.